File: easy/read_more.lua
for line in io.lines(arg[1]) do
  if #line > 55 then
    local i, p = 0, 41
    while true do
      i = string.find(line, " ", i+1)
      if not i or i > 40 then break end
      p = i
    end
    line = line:sub(1, p-1) .. "... <Read More>"
  end
  print(line)
end

File: easy/read_more.hs
import System.Environment (getArgs)
import Data.List (elemIndices)

readMore  :: String -> String
readMore s | length s < 56 = s
           | otherwise     = take (last e) s ++ "... <Read More>"
           where e = 40 : [x | x <- elemIndices ' ' s, x < 40]

main :: IO ()
main = do
    [inpFile] <- getArgs
    input <- readFile inpFile
    putStr . unlines . map readMore $ lines input
